src/proposals/20251002/SparkEthereum_20251002.t.sol

```diff
@@ -0,0 +1,352 @@
+
+        ReserveConfig memory usdc = _findReserveConfigBySymbol(allConfigsBefore, 'USDC');
+        ReserveConfig memory usdt = _findReserveConfigBySymbol(allConfigsBefore, 'USDT');
+
+        assertEq(usdc.reserveFactor, 10_00);
+        assertEq(usdt.reserveFactor, 10_00);
+
+        executeAllPayloadsAndBridges();
+
+        ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot('', ctx.pool);
+
+        usdc.reserveFactor = 1_00;
+        usdt.reserveFactor = 1_00;
+
+        _validateReserveConfig(usdc, allConfigsAfter);
+        _validateReserveConfig(usdt, allConfigsAfter);
+    }
+
+    function test_ETHEREUM_sparkLend_lbtcCapAutomatorUpdates() public onChain(ChainIdUtils.Ethereum()) {
+        _assertSupplyCapConfig(Ethereum.LBTC, 2500, 250, 12 hours);
+
+        executeAllPayloadsAndBridges();
+
+        _assertSupplyCapConfig(Ethereum.LBTC, 10_000, 500, 12 hours);
+    }
+
+    function test_ETHEREUM_sparkVaultsV2_configureSPUSDC() public onChain(ChainIdUtils.Ethereum()) {
+        _testVaultConfiguration({
+            vault_:     Ethereum.SPARK_VAULT_V2_SPUSDC,
+            minVsr:     1e27,
+            maxVsr:     TEN_PCT_APY,
+            depositCap: 50_000_000e6,
+            amount:     1_000_000e6
+        });
+    }
+
+    function test_ETHEREUM_sparkVaultsV2_configureSPUSDT() public onChain(ChainIdUtils.Ethereum()) {
+        _testVaultConfiguration({
+            vault_:     Ethereum.SPARK_VAULT_V2_SPUSDT,
+            minVsr:     1e27,
+            maxVsr:     TEN_PCT_APY,
+            depositCap: 50_000_000e6,
+            amount:     1_000_000e6
+        });
+    }
+
+    function test_ETHEREUM_sparkVaultsV2_configureSPETH() public onChain(ChainIdUtils.Ethereum()) {
+        _testVaultConfiguration({
+            vault_:     Ethereum.SPARK_VAULT_V2_SPETH,
+            minVsr:     1e27,
+            maxVsr:     FIVE_PCT_APY,
+            depositCap: 10_000e18,
+            amount:     1_000e18
+        });
+    }
+
+    function _testVaultConfiguration(
+        address vault_,
+        uint256 minVsr,
+        uint256 maxVsr,
+        uint256 depositCap,
+        uint256 amount
+    ) internal {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+
+        ISparkVaultV2 vault = ISparkVaultV2(vault_);
+
+        bytes32 takeKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(Ethereum.ALM_CONTROLLER).LIMIT_SPARK_VAULT_TAKE(),
+            vault_
+        );
+        bytes32 transferKey = RateLimitHelpers.makeAssetDestinationKey(
+            MainnetController(Ethereum.ALM_CONTROLLER).LIMIT_ASSET_TRANSFER(),
+            vault.asset(),
+            vault_
+        );
+
+        assertEq(vault.hasRole(vault.DEFAULT_ADMIN_ROLE(), Ethereum.SPARK_PROXY), true);
+        assertEq(vault.hasRole(vault.SETTER_ROLE(), Ethereum.ALM_OPS_MULTISIG),   false);
+        assertEq(vault.hasRole(vault.TAKER_ROLE(), Ethereum.ALM_PROXY),           false);
```

Comment:
Always align all params of functions that are grouped together, so align params of assertEq, but also align params of hasRole inside:

```solidity
        assertEq(vault.hasRole(vault.DEFAULT_ADMIN_ROLE(), Ethereum.SPARK_PROXY),      true);
        assertEq(vault.hasRole(vault.SETTER_ROLE(),        Ethereum.ALM_OPS_MULTISIG), false);
        assertEq(vault.hasRole(vault.TAKER_ROLE(),         Ethereum.ALM_PROXY),        false);
```

Reply:
Use getRoleMemberCount for all three below these three and assert they are all 1,0,0 then 1,1,1 (example of function call below)

<img width="640" height="110" alt="Image" src="https://github.com/user-attachments/assets/847b5380-85ff-46a5-95fd-65a378b3a4a0" />